app/src/main/java/org/mozilla/fenix/components/toolbar/EngineViewTopBehavior.kt

```diff
@@ -0,0 +1,46 @@
+package org.mozilla.fenix.components.toolbar
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import androidx.coordinatorlayout.widget.CoordinatorLayout
+import mozilla.components.concept.engine.EngineView
+import mozilla.components.support.ktx.android.view.findViewInHierarchy
+
+/**
+ * A [CoordinatorLayout.Behavior] implementation to be used with [EngineView] when placing a toolbar at the
+ * bottom of the screen.
+ *
+ * Using this behavior requires the toolbar to use the BrowserToolbarTopBehavior.
+ *
+ * This implementation will update the vertical clipping of the [EngineView] so that top-aligned web content will
+ * be drawn above the native toolbar.
+ */
+class EngineViewTopBehavior(
+    context: Context?,
+    attrs: AttributeSet?
+) : CoordinatorLayout.Behavior<View>(context, attrs) {
+    @SuppressLint("LogUsage")
+    override fun layoutDependsOn(parent: CoordinatorLayout, child: View, dependency: View): Boolean {
+        // This package does not have access to "BrowserToolbar" ... so we are just checking the class name here since
+        // we actually do not need anything from that class - we only need to identify the instance.
+        // Right now we do not have a component that has access to (concept/browser-toolbar and concept-engine).
+        // Creating one just for this behavior is too excessive.
+        if (dependency::class.java.simpleName == "BrowserToolbar") {
+            return true
+        }
+
+        return super.layoutDependsOn(parent, child, dependency)
+    }
+
+    /**
+     * Apply vertical clipping to [EngineView]. This requires [EngineViewTopehavior] to be set
+     * in/on the [EngineView] or its parent. Must be a direct descending child of [CoordinatorLayout].
+     */
+    override fun onDependentViewChanged(parent: CoordinatorLayout, child: View, dependency: View): Boolean {
+        val engineView = child.findViewInHierarchy { it is EngineView } as EngineView?
+        engineView?.setVerticalClipping(dependency.height - dependency.translationY.toInt())
```

Comment:
This looks pretty odd to me.  This `setVerticalClipping` API ends up [calling `setFixedBottomOffset` in GeckoView](https://searchfox.org/mozilla-central/rev/652014ca1183c56bc5f04daf01af180d4e50a91c/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/GeckoDisplay.java#140-144), the function does just set a margin for position:fixed elements at bottom of viewport. As far as I can tell, we need at least one new API make the top toolbar dynamic properly, the API would be a function to set an offset of the root viewport (i.e. not only for position:fixed elements).

CCing @pocmo and @snorp.

Reply:
Yes, for the top toolbar this should not be called at all. And I think @sblatz decided to not use this behavior implementation for the top toolbar, right? Afaik this was never really used in Fenix.